geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -967,7 +981,8 @@ private void checkForStuckKeys() {
             tmpSel.selectNow(); // clear canceled key
             sc.registerWithSelector2(selector);
           } else {
-            if (tmpsk.isValid() && tmpsk.isReadable()) {
+            if (tmpsk.isValid() && (tmpsk.isReadable() || tmpsk.isWritable())) {
```

Comment:
Why did you add isWritable? I think the old code only used non-blocking on reads. Are you also now doing non-blocking writes?
Note this common expression (line 984 and 1129) could be extracted into a method

Reply:
Without this, test are failing, and no connection can not be established.